edx_proctoring/tests/test_services.py

```diff
@@ -239,3 +259,23 @@ def undo_override_subsection_grade(self, user_id, course_key_or_id, usage_key_or
     def should_override_grade_on_rejected_exam(self, course_key):
         """Mock will always return instance variable: rejected_exam_overrides_grade"""
         return self.rejected_exam_overrides_grade
+
+
+class MockCertificateService(object):
+    """
+    mock Certificate Service
+    """
+    def __init__(self):
+        """
+        Initialize empty data stores for generated certificate
+        """
+        self.generated_certificate = {}
+
+    def invalidate_certificate(self, user_id, course_key_or_id):
+        """
+        Get the generated certificate for key (user_id + course_key) and invalidate certificate
+        when dropped below passing threshold due to suspicious proctored exam
+        """
+        key = str(user_id) + str(course_key_or_id)
+        self.generated_certificate[key] = MockGeneratedCertificate()
+        self.generated_certificate[key].mock_invalidate()
```

Comment:
`mock_invalidate` method simply override all  fields of certificate.  I didn't get how we are testing certificate invalidation here. Lets meet in person and discuss this.

Reply:
This is just a mock service. We still need a test in [`test_api.py`](https://github.com/edx/edx-proctoring/blob/master/edx_proctoring/tests/test_api.py) checking that when a grade is overridden this invalidate_certificate function is called.